config/crds/resources/apiextensions.k8s.io_v1_customresourcedefinition_iamauditconfigs.iam.cnrm.cloud.google.com.yaml

```diff
@@ -81,12 +82,39 @@ spec:
               resourceRef:
                 description: Immutable. Required. The GCP resource to set the IAMAuditConfig
                   on (e.g. project).
+                oneOf:
+                - not:
+                    required:
+                    - external
+                  required:
+                  - name
+                - not:
+                    anyOf:
+                    - required:
+                      - name
+                    - required:
+                      - namespace
+                  required:
+                  - external
+                - not:
```

Comment:
I understand that you are probably copying the schema over from IAMPolicy/IAMPartialPolicy, but the following rule doesn't seem to make much sense to me.

```
                - not:
                    anyOf:
                    - required:
                      - name
                    - required:
                      - namespace
                    - required:
                      - apiVersion
                    - required:
                      - external
```
This basically means "you shouldn't specify any of `name`, `namespace`, `apiVersion`, or `external` field". I don't know if this is a valid case for resourceRef here.

Can we hold on adding the oneOf rule?

Reply:
I also got a bit confused by this at first but I think it means that `kind` by itself is ok but not any of {name, namespace, apiVersion, external} by themselves only.

Reply:
Yeah, I'm not sure whether `kind` by itself is okay or not though. Have you identified such edge cases?